### PR TITLE
Elf plugin code cleaning ##refactor

### PR DESCRIPTION
**Pull request by ret2libc**

### CODE REVIEW

File: libr/bin/format/elf/elf.c [hunk 1]
```diff
@@ -185,61 +253,19 @@ static int init_phdr(ELFOBJ *bin) {
...
-	}
-	/* Here is the where all the fun starts.
-	 * Linux kernel since 2005 calculates phdr offset wrongly
-	 * adding it to the load address (va of the LOAD0).
-	 * See `fs/binfmt_elf.c` file this line:
-	 *    NEW_AUX_ENT(AT_PHDR, load_addr + exec->e_phoff);
-	 * So after the first read, we fix the address and read it again
-	 */
-	if (linux_kern_hack && phdr_found) {
-		ut64 load_addr = Elf_(r_bin_elf_get_baddr) (bin);
```
[the remark this answers is not in the document]
Comment (ret2libc) [reply libr/bin/format/elf/elf.c]: may be the comment it's a bit misleading, but it's not a problem in the kernel AFAIK. It's in the way some kernels loaded ELF files with PT_PHDR phdr.

**Comment by ret2libc** [comment]
I would like to merge this as soon as we release 3.1, so we have much more time to see if it doesn't break. I'll keep this on hold for another week

**Comment by radare** [comment]
I have some more issues in elf and i want to merge this too asap. Just wanting to know if u tested a kernel. If it works then its fine to merge

**Comment by ret2libc** [comment]
Merge then. But no more risky merging after this I'd say, there's just 1 week before release.

**Pull request merged**